Examples/Scripts/Python/MLAmbiguityResolution/ambiguity_solver_perf.py

```diff
@@ -0,0 +1,138 @@
+import glob
+import os
+import math
+
+import pandas as pd
+import numpy as np
+
+from sklearn.preprocessing import LabelEncoder, StandardScaler, OrdinalEncoder
+from sklearn.cluster import DBSCAN
+
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.utils
+
+import ast
+
+
+def PrepareDataSet(CKS_files: list[str]) -> pd.DataFrame:
+    """Read the dataset from the different file, remove the pure duplicate tracks and combine the datasets"""
+    """
+    @param[in] CKS_files: DataFrame contain the data from each track files (1 file per events usually)
+    @return: combined DataFrame containing all the track, ordered by events and then by truth particle ID in each events 
+    """
+    globalindex = 0
+    data = []
+    for f in CKS_files:
+        datafile = pd.read_csv(f)
+        # Remove tracks with less than 7 measurements
+        datafile = datafile[datafile["nMeasurements"] > 6]
+        datafile = datafile.sort_values("good/duplicate/fake", ascending=False)
+        # Remove pure duplicate (tracks purely identical) keep the ones good one if among them.
+        datafile = datafile.drop_duplicates(
+            subset=[
+                "particleId",
+                "Hits_ID",
+                "nOutliers",
+                "nHoles",
+                "nSharedHits",
+                "chi2",
+            ],
+            keep="first",
+        )
+        datafile = datafile.sort_values("particleId")
+        # Set truth particle ID as index
+        datafile = datafile.set_index("particleId")
+        # Transform the hit list from a string to an actual list
+        hitsIds = []
+        mergedIds = []
+        for list in datafile["Hits_ID"].values:
+            hitsIds.append(ast.literal_eval(list))
+        datafile["Hits_ID"] = hitsIds
+        # Combine dataset
+        data.append(datafile)
+    return data
```

Comment:
looks like this function is duplicated in `train_ambiguity_solver.py`. would be cool if we could import it in one of the modules

Reply:
Good idea ! I will see what I can do

Reply:
I have done the best I could in that regard, it should be clearer with the new implementation